contracts/ERC165.sol

```diff
@@ -0,0 +1,20 @@
+pragma solidity ^0.4.23;
+
+
+/**
+ * @title ERC165
+ * @dev https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md
+ */
+interface ERC165 {
+
+  /**
+   * @notice Query if a contract implements an interface
+   * @param _interfaceId The interface identifier, as specified in ERC-165
+   * @dev Interface identification is specified in ERC-165. This function
+   * @dev  uses less than 30,000 gas.
```

Comment:
Question here: what happens if there are too many methods and interfaces and this ends up consuming more than 30000 gas?

Reply:
The lookup will always use less than 30k gas; the gas limit would come in if we're doing some special calculations with a bunch of branches in the code, I think.